### ISSUE TITLE
Question on the test results on my own data and the training dataset choices

### ISSUE THREAD

**Issue by SimonSongg** [hlwang1124/SNE-RoadSeg#24]
Hello, I tried to use the data captured by ZED 2 depth camera ,which are shown below, to obtain the road prediction. But the network doesn't work at all. Do you have any ideas on the results? And I noticed that in your paper, three datasets are used to train the model, and the code you show here is only for KITTI dataset? Thank you in advance!
![1](https://user-images.githubusercontent.com/70959938/110279270-cdc66f80-8013-11eb-8f90-1bd539e5b5e7.png)
![normal](https://user-images.githubusercontent.com/70959938/110279277-d028c980-8013-11eb-8e66-1fa0d5ec9603.png)
![pred](https://user-images.githubusercontent.com/70959938/110279278-d0c16000-8013-11eb-9b8b-bcaaf0ee2087.png)
![prob_map](https://user-images.githubusercontent.com/70959938/110279282-d159f680-8013-11eb-936e-ef852a2c4948.png)
![rgb](https://user-images.githubusercontent.com/70959938/110279283-d159f680-8013-11eb-873c-d95f7b35e571.png)

**Comment by SimonSongg** [comment]
Does it result from the grainy texture in the depth map which is obvious in the SNE map? I may try different depth modes to test it. I am thinking if intel realsense d455 will have better performance...

**Comment by hlwang1124** [comment]
Hi, @SimonSongg . To use your own dataset, the camera intrinsic parameter `camParam` needs to be updated. I attach the dataloader of our R2D dataset [here](https://github.com/hlwang1124/SNE-RoadSeg/files/6134195/R2D_dataset.zip) for your reference. You can update your own dataloader accordingly.

In addition, I have tried using RealSense D415 RGB-D camera for the drivable area and road anomaly detection, as shown in [here](https://github.com/hlwang1124/NIM). Please note that the normal estimation module does not have training parameters, but the segmentation part does. Although there exist invalid pixels in the RealSense depth image, the corresponding normal results seem better than yours and are enough for the segmentation task. However, I re-trained the network using the RealSense data for the segmentation task, and I am not sure if it can work based on the model trained on the KITTI dataset.

In summary, I suggest you first try to update the camera intrinsic parameter and test the performance. If it still does not work well, maybe you need to try another depth camera or re-train the network.

**Comment by SimonSongg** [comment]
Thanks for your reply. Yes I used my own camparam. Do you mean you train the network with only one dataset at a time (either KITTI or R2D)? Yeah the depth from your D415 looks better. ZED 2 RGBD camera provide several depth mode, I will try to capture with different mode. So you have tested the data from d415 and got a good segmentation result with re-trainning seg-network? If it is true, then maybe I need to get a realsense camera cuz ZED itself is also computing consuming...

**Comment by hlwang1124** [comment]
Yes. Good luck:)

**Comment by SimonSongg** [comment]
![rgb](https://user-images.githubusercontent.com/70959938/111023970-12d51200-8417-11eb-9fce-919d666c2bbd.png)
![1](https://user-images.githubusercontent.com/70959938/111023972-15d00280-8417-11eb-9631-1c2031e8f5d9.png)
![normal](https://user-images.githubusercontent.com/70959938/111023973-16689900-8417-11eb-8e99-f32b08d9d7e0.png)
![pred](https://user-images.githubusercontent.com/70959938/111023974-17012f80-8417-11eb-9430-8d26f5400d7b.png)
![prob_map](https://user-images.githubusercontent.com/70959938/111023975-1799c600-8417-11eb-9db6-65a94613c463.png)

Hi, @hlwang1124 . I tried a different depth mode and the normal estimation looks similar to the one you captured by d415, but still bad results in prediction. Might need to re-train the model?

**Comment by hlwang1124** [comment]
Yes. It seems that the generalization ability is not that good.

**Comment by SimonSongg** [comment]
![prob_map](https://user-images.githubusercontent.com/70959938/111024399-968ffe00-8419-11eb-8380-82d6e186c7a4.png)
![rgb](https://user-images.githubusercontent.com/70959938/111024402-9859c180-8419-11eb-8720-e7e5d2263768.png)
![1](https://user-images.githubusercontent.com/70959938/111024404-98f25800-8419-11eb-9eea-6e00d50b6cf8.png)
![normal](https://user-images.githubusercontent.com/70959938/111024405-998aee80-8419-11eb-8395-757e4aa66577.png)
![pred](https://user-images.githubusercontent.com/70959938/111024406-9a238500-8419-11eb-9054-4e4308882591.png)

This is another mode results. Looks better.

**Comment by SimonSongg** [comment]
At least the road can be recognized in the probability map lol
